code/src/OSFramework/Interface/IDataSource.ts

```diff
@@ -59,6 +59,8 @@ namespace OSFramework.Grid {
          */
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         getProviderDataSource(): any;
+
+        getRowNumberByKey(key: string): number;
```

Comment:
Can you please add a comment here so that we have a little help while using the method in the future?